### PR TITLE
fix: create dirs before running rattler-build

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

**Comment by wolfv** [comment]
When this one is merged and released, I'll try to remvoe the create dirs stuff from pixi again.

File: crates/pixi-build/src/bin/pixi-build-rattler-build/rattler_build.rs [hunk 1]
```diff
@@ -200,6 +204,8 @@ impl Protocol for RattlerBuildBackend {
     }
 
     async fn build_conda(&self, params: CondaBuildParams) -> miette::Result<CondaBuildResult> {
+        fs::create_dir_all(&params.work_directory).into_diagnostic()?;
```
Comment (nichmor) [inline crates/pixi-build/src/bin/pixi-build-rattler-build/rattler_build.rs:207]:
```suggestion
        // Create the work directory if it does not exist
        fs::create_dir_all(&params.work_directory).into_diagnostic()?;
```

Comment (nichmor) [reply crates/pixi-build/src/bin/pixi-build-rattler-build/rattler_build.rs:207]: to keep them the same

**Review by nichmor — COMMENTED** [review]
have you used something to sort pixi.toml?
let's add it in pre-commit!

**Review by nichmor — APPROVED** [review]

**Comment by wolfv** [comment]
I didn't use anything, just did it by hand and VS Code sort lines.

**Pull request merged**